docs/docs/reference/state-representation.md

```diff
@@ -0,0 +1,1757 @@
+variables to a unit random state within a uniform distribution. As with the regular constructor, a vector of joint names
+or an integer number of joints can be supplied.
+
+<Tabs groupId="programming-language">
+<TabItem value="c++" label="C++">
+
+```cpp
+// initialize the joint state to zero values
+state_representation::JointState::Zero("my_robot", joint_names);
+state_representation::JointState::Zero("my_robot", 3);
+
+// initialize the joint state to random values
+state_representation::JointState::Random("my_robot", joint_names);
+state_representation::JointState::Random("my_robot", 3);
+```
+
+</TabItem>
+</Tabs>
+
+### Joint getters and setters
+
+Each state variable has a corresponding getter and setter to access or modify the data after construction.
+
+The following groups of variables can be accessed or set as a vector ordered by the joint indexes / names:
+
+- `get_positions()`, `set_positions({...})` in radians
+- `get_velocities()`, `set_velocities({...})` in radians per second
+- `get_accelerations()`, `set_accelerations({...})` in radians per second squared
+- `get_torques()`, `set_torques({...})` in Newton-meters
+
+The vector setters are defined for both `Eigen::VectorXd` and `std::vector<double>`:
+
+<Tabs groupId="programming-language">
+<TabItem value="c++" label="C++">
+
+```cpp
+state_representation::JointState js("my_robot", 3);
+js.set_positions(Eigen::Vector3d(.5, 1., 0.));
+js.set_positions(std::vector<double>{.5, 1., 0.});
+```
+
+</TabItem>
+</Tabs>
+
+When setting a vector of state variables, the size of the input vector must match the number of joints.
+
+<Tabs groupId="programming-language">
+<TabItem value="c++" label="C++">
+
+```cpp
+js.set_positions(Eigen::Vector4d::Random()); // will throw an IncompatibleSizeException
+```
+
+</TabItem>
+</Tabs>
+
+The state variable values of each individual joint can also be accessed as scalars, using either the integer joint
+index or the string joint name as the identifier:
+
+- `get_position(id)`, `set_position(x, id)` in radians
+- `get_velocity(id)`, `set_velocity(x, id)` in radians per second
+- `get_acceleration(id)`, `set_acceleration(x, id)` in radians per second squared
+- `get_torque(id)`, `set_torques(x, id)` in Newton-meters
+
+<Tabs groupId="programming-language">
+<TabItem value="c++" label="C++">
+
+```cpp
+auto js = state_representation::JointState::Random("my_robot", { "hip", "knee" });
+
+js.get_position(0); // get the position of the hip joint
+js.get_velocity("hip"); // get the velocity of the hip joint
+js.set_acceleration(0.5, 1); // set the acceleration of the knee joint to 5 rad/s^2
+js.set_torque(2.0, "knee"); // set the torque of the knee joint to 2 Nm
+```
+
+</TabItem>
+</Tabs>
+
+### Joint state addition, subtraction and scaling
+
+Two `JointState` objects can be combined with addition or subtraction, provided they have the same name and joint names.
```

Comment:
This is actually not correct, they don't need the same name:
https://github.com/aica-technology/control-libraries/blob/1f6c10fd4b4b5d3e9e35ce67a10a3dc8931ba865/source/state_representation/src/space/joint/JointState.cpp#L437
however, the error message is wrong: https://github.com/aica-technology/control-libraries/blob/1f6c10fd4b4b5d3e9e35ce67a10a3dc8931ba865/source/state_representation/src/space/joint/JointState.cpp#L519